mypyc/ops_misc.py

```diff
@@ -281,6 +281,14 @@
     error_kind=ERR_NEVER,
     emit=simple_emit('{dest} = Py_TYPE({args[0]}) == (PyTypeObject *){args[1]};'))
 
+# C-level "or" operator
+fast_or_op = custom_op(
+    name='fast_or',
+    arg_types=[bool_rprimitive, bool_rprimitive],
+    result_type=bool_rprimitive,
+    error_kind=ERR_NEVER,
+    emit=simple_emit('{dest} = {args[0]} || {args[1]};'))
```

Comment:
If the goal of this is to have short-circuiting behavior, it won't work as intended. Both of the arguments will be evaluated and assigned to temporary variables before this is executed.

Reply:
I initially wanted to have short-circuiting, but I am not sure this will give some visible win: in most cases we are trading few pointer comparisons for a few `goto`s. If you think it is important, I can try rewriting this.

I also want to skip any boxing/unboxing of `bool`s, will the logic in `shortcircuit_helper` guarantee this?

Reply:
I'm not sure if it will be important or not? Might be worth benchmarking, is definitely worth a comment making clear the situation, I think.

I think that `shortcircuit_helper` won't generate any spurious boxing/unboxing.

Reply:
> won't generate any spurious boxing/unboxing

Yes, I tried and I don't see anything bad in IR. I will try benchmarking now.

Reply:
It is hard to see the difference, maybe 10-20ms. Still I would keep it, because it makes the logic cleaner.